client/me/purchases/controller.jsx

```diff
@@ -156,7 +156,7 @@ export default {
 		);
 	},
 
-	list() {
+	list( context ) {
```

Comment:
>  The notice message would live in PurchasesController in this example, yeah?

Yes, this is where you would initialise it before redirecting to the expected clean url.
It's also highly possible that "global" notices don't fit in the current design. That should be checked before. Anyway this looks rather like exercise for v2 ;) I hope we can get back to it in the future.

In the PHP world this behavior is called Flash Messages. 
Symfony 3 example: http://symfony.com/doc/current/book/controller.html#flash-messages.
Zend Framework 2 example: http://framework.zend.com/manual/current/en/modules/zend.view.helpers.flash-messenger.html